lib/rmt/mirror.rb

```diff
@@ -20,6 +20,26 @@ def initialize(mirroring_base_dir: RMT::DEFAULT_MIRROR_DIR, logger:, mirror_src:
     )
   end
 
+  def mirror_suma_product_tree(repository_url:)
+    # we have an inconsistency in how we mirror in offline mode
+    # in normal mode we mirror in the following way:
+    # base_dir/repo/...
+    # however, in offline mode we mirror in the following way
+    # base_dir/...
+    # we need this extra step to ensure that we write to the public directory
+    base_dir = @mirroring_base_dir
+    base_dir = File.expand_path(File.join(@mirroring_base_dir, '/../')) if @mirroring_base_dir == RMT::DEFAULT_MIRROR_DIR
+
+    @repository_dir = File.join(base_dir, '/suma/')
+    @downloader.repository_url = URI.join(repository_url)
+    @downloader.destination_dir = @repository_dir
+
+    @logger.info _('Mirroring SUMA product tree to %{dir}') % { dir: @repository_dir }
+    @downloader.download('product_tree.json')
+  rescue RMT::Downloader::Exception => e
+    raise RMT::Mirror::Exception.new(_('Could not mirror suma product tree with error: %{error}') % { error: e.message })
```

Comment:
Can we not just log the error here and remove all try/catch statements in the other files?

Reply:
I guess I feel not having the function a black box and that we let the cli itself handle the failure state.

Also, it should make solving issues like [this](https://github.com/SUSE/rmt/issues/334) easier. I think that we actually want to implement a summary of failures these cli commands.